app/domain/authentication/authn_jwt/restriction_validation/validate_restrictions_one_to_one.rb

```diff
@@ -4,21 +4,34 @@ module RestrictionValidation
       # This class is responsible for retrieving the correct value from the JWT token
       # of the requested attribute.
       class ValidateRestrictionsOneToOne
-        def initialize(decoded_token:)
+        def initialize(decoded_token:, mapping_claims:)
           @decoded_token = decoded_token
+          @mapping_claims = mapping_claims
         end
 
         def valid_restriction?(restriction)
-          restriction_name = restriction.name
+          annotation_name = restriction.name
+          claim_name = @mapping_claims.fetch(annotation_name, annotation_name)
```

Comment:
I think we need a debug log here to know what we are validating the mapped value or the origin, WDYT?

Reply:
```
CONJ00125D Parsing mapping claims value 'branch:ref'
CONJ00118D Validating claim name 'branch'
CONJ00117D Claims denylist value is '["iss", "exp", "nbf", "iat", "jti", "aud"]'
CONJ00119D Successfully validated claim name 'branch'
CONJ00118D Validating claim name 'ref'
CONJ00117D Claims denylist value is '["iss", "exp", "nbf", "iat", "jti", "aud"]'
CONJ00119D Successfully validated claim name 'ref'
CONJ00127D Mapping annotation name 'branch' to the claim name 'ref'
CONJ00126D Successfully parsed mapping claims '{"branch"=>"ref"}'
CONJ00130I Successfully fetched mapping claims '{"branch"=>"ref"}'
CONJ00046D Validating resource restriction for role 'host/myapp'
CONJ00032I Extracting resource restrictions for authenticator 'authn-jwt' from host 'host/myapp'
CONJ00024I Retrieved value of annotation 'authn-jwt/raw/project-id'
CONJ00024I Retrieved value of annotation 'authn-jwt/raw/branch'
CONJ00040D Resource restrictions were extracted: '["project-id", "branch"]'
CONJ00041D Validating resource restrictions configuration
CONJ00042D Resource restrictions configuration validated
CONJ00044D Validating resource restrictions on request
CONJ00048D Validating resource restriction on request: 'project-id'
CONJ00048D Validating resource restriction on request: 'branch'
CONJ00045D Resource restrictions matched request
CONJ00030D Resource restrictions validated
CONJ00103D 'validate_restrictions' passed successfully
```
Do you think that we need one another log message about mapping?

Reply:
Yes (i really like logs  😂  )
I think after the log message `CONJ00048D `
If there is a mapping, what is actual value we are validation against the token